examples/self-diagnostics/src/main.rs

```diff
@@ -92,41 +23,43 @@ fn init_meter_provider() -> opentelemetry_sdk::metrics::SdkMeterProvider {
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
-    let logger_provider = init_logger_provider();
+    // OpenTelemetry uses `tracing` crate for its internal logging. Unless a
+    // tracing subscriber is set, the logs will be discarded. In this example,
+    // we configure a `tracing` subscriber to:
+    // 1. Print logs of level INFO or higher to stdout.
+    // 2. Filter logs from OpenTelemetry's dependencies (like tonic, hyper,
+    // reqwest etc. which are commonly used by the OTLP exporter) to only print
+    // ERROR-level logs. This filtering helps reduce repetitive log messages
+    // that could otherwise create an infinite loop of log output. This is a
+    // workaround until
+    // https://github.com/open-telemetry/opentelemetry-rust/issues/761 is
+    // resolved.
+
+    // Target name used by OpenTelemetry always start with "opentelemetry".
+    // Hence, one may use "add_directive("opentelemetry=off".parse().unwrap())"
+    // to turn off all logs from OpenTelemetry.
+
+    let filter = EnvFilter::new("info")
+        .add_directive("hyper=error".parse().unwrap())
+        .add_directive("tonic=error".parse().unwrap())
+        .add_directive("h2=error".parse().unwrap())
+        .add_directive("tower=error".parse().unwrap())
+        .add_directive("reqwest=error".parse().unwrap());
```

Comment:
we can remove this filter if none of these transport dependencies are used in the example.

Reply:
let me move it to the otlp examples, so when users face this issue, they have something to easily refer to, to mitigate the infinite logs problem.

Reply:
will add this shortly in a follow up PR.